### DISCUSSION TITLE
How could I install specific commit version of Hyprland in Nixos

### DISCUSSION THREAD

**Discussion by lVentus** [hyprwm/Hyprland#6080] [Q&A]
wiki taught how to install dev version, but how can I set to specific commit? one of my plugins works only on one commit.

and another question, when I use Hyprland, when I click something in any software, the cursor will move to the center of the window. and waybar will also init again. what happened?

**Comment by fufexan** [comment]
If you're using the flake, pin the revision by adding `&rev=<commit>` in your `inputs.hyprland.url`.

> and another question, when I use Hyprland, when I click something in any software, the cursor will move to the center of the window. and waybar will also init again. what happened?

No idea about it. Best to search the issues and open one if you can't find anything.

**Comment by drxm1** [comment]
As fufexan noted the commit can be pinned.
Here is an example of how I do it currently (in flake.nix inputs), which is working:

```
hyprland = {
      type = "git";
      url = "https://github.com/hyprwm/Hyprland";
      rev = "882f7ad7d2bbfc7440d0ccaef93b1cdd78e8e3ff"; # Full sha256 of commit, choose a tagged version for plugin compatibility
      submodules = true;
      inputs.nixpkgs.follows = "nixpkgs";
};
```

**Reply by fufexan** [reply]
Submodules are no longer needed. `type` can be changed to `github`.
